org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/XtextEditor.java

```diff
@@ -327,6 +328,14 @@ public void doSave(IProgressMonitor progressMonitor) {
 		callback.afterSave(this);
 	}
 	
+	@Override
+	protected void performSave(boolean overwrite, IProgressMonitor progressMonitor) {
+		boolean unsyncOverwrite = !((XtextDocumentProvider)getDocumentProvider()).isSynchronized(getEditorInput());
```

Comment:
Preferably use an instance-of guarded cast to `org.eclipse.ui.texteditor.IDocumentProviderExtension3.isSynchronized(Object)` for the unlikely though possible case where the document provider is overridden

Reply:
IResource.isSynchornized() doesn't have the same understanding of FileDocumentProvider.checkSynchronizationState() of synchronzied.. If XtextDocumentProvider.isSynchronized is called multiple times, it will be for the first time unsynchron (return false) and for the next calls it will IResource.isSynchronized return true. But isSynchronized() on FileDocumentProvider will be false.. So I've to trigger it and don't care about the result.

Reply:
done.